src/App.js

```diff
@@ -1,5 +1,133 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+
 class App {
-  async run() {}
+  run = async () => {
+    try {
+      const input = await MissionUtils.Console.readLineAsync("덧셈할 문자열을 입력해 주세요.\n");
+
+      if (input.length === 0) {
+        MissionUtils.Console.print('결과 : 0');
+        return;
+      }
+
+      // 커스텀 구분자가 있는지 검사 후 배열을 반환
+      const { inputArr, delimiters } = this.parseInputToArray(input);
+
+      // 유효한 입력값일 경우, 계산하는 메서드를 실행
+      if (this.isValid(inputArr, delimiters)) {
+        this.calculate(inputArr);
+      } else {
+        throw new Error('[ERROR] 잘못된 구분자를 사용하셨습니다.');
+      }
+    } catch (error) {
+      MissionUtils.Console.print(error.message);
+      // 에러가 발생했을 때 Promise가 reject되도록 에러를 다시 throw
+      throw error;
+    }
+  }
+
+  isValid = (inputArr, delimiters) => {
+    const isValidInput = inputArr.every((str, idx) => {
+      if (idx % 2 === 0) { 
+        // 숫자인지 확인
+        if (isNaN(parseFloat(str)) || !isFinite(str)) {
+          throw new Error(`[ERROR] 유효하지 않은 숫자입니다: ${str}`);
+        }
+        return true;
+      } else { 
+        // 구분자가 올바른지 확인
+        if (!delimiters.includes(str)) {
+          throw new Error(`[ERROR] 유효하지 않은 구분자입니다: ${str}`);
+        }
+        return true;
+      }
+    });
+  
+    return isValidInput; 
+  }
+
+  // 입력받은 input을 배열로 바꿔주는 메서드
+  parseInputToArray = (input) => {
+    const delimiters = [',', ':'];
+
+    // 커스텀 구분자가 있는지 확인
+    if (input.slice(0, 2) === '//') {
+      for (let i = 2; i < input.length; i++) {
+        if (input[i] === "\\" && input[i + 1] === "n") {
+          const NEW_DELIMITER = input.slice(2, i);
+          delimiters.push(NEW_DELIMITER);
+          input = input.slice(i + 2);
+          break;
+        }
+      }
+    }
+
+    let temp = '';
+    const result = [];
+    for (let i = 0; i < input.length; i++) {
+      const char = input[i];
+    
+      // char가 숫자인 경우
+      if (typeof Number(char) === 'number' && !isNaN(Number(char)) && char !== ' ') {
```

Comment:
`isNaN`은 내부에서 `Number`로 바꾼 후 `NaN`인지 판단해주는 메서드이기 때문에 `isNaN` 안에서 `Number`로 한번 더 감싸지 않아도 될 것 같습니다 ☺️
```suggestion
      if (typeof Number(char) === 'number' && !isNaN(char) && char !== ' ') {
```
참고 : [isNaN의 MDN문서](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/isNaN#%ED%98%BC%EB%9E%80%EC%8A%A4%EB%9F%AC%EC%9A%B4_%ED%8A%B9%EB%B3%84_%EC%BC%80%EC%9D%B4%EC%8A%A4_%ED%96%89%EB%8F%99) 

그리고 `char !== ' '`는 어떤 이유로 조건문에 넣으신건지 궁금합니다!